service_container/injection_types.rst

```diff
@@ -100,6 +100,113 @@ working with optional dependencies. It is also more difficult to use in
 combination with class hierarchies: if a class uses constructor injection
 then extending it and overriding the constructor becomes problematic.
 
+Immutable-setter Injection
+--------------------------
+
+.. versionadded:: 4.3
+
+    The ``immutable-setter`` injection was introduced in Symfony 4.3.
+
+Another possible injection is to use a method which returns a separate instance
+by cloning the original service, this approach allows you to make a service immutable::
+
+    // ...
+    use Symfony\Component\Mailer\MailerInterface;
+
+    class NewsletterManager
+    {
+        private $mailer;
+
+        /**
+         * @required
+         * @return static
+         */
+        public function withMailer(MailerInterface $mailer)
+        {
+            $new = clone $this;
+            $new->mailer = $mailer;
+
+            return $new;
+        }
+
+        // ...
+    }
+
+In order to use the full potential of this approach, you use a third argument
```

Comment:
We didn't introduce setters yet, telling about the third argument is too early I think.
The sentence also is strange, the "full potential" is unclear.
Let's move this section after Setter Injection, and adjust (I'll review again after :) )